File: 1260. Shift 2D Grid.java
class Solution {
    public List<List<Integer>> shiftGrid(int[][] grid, int k) {

        int row = grid.length;
        int column = grid[0].length;
        int total = row * column;

        k %= total;

        List<List<Integer>> answer = new ArrayList<>();

        for (int i = 0; i < row; i++) {
            List<Integer> temp = new ArrayList<>();

            for (int j = 0; j < column; j++) {
                temp.add(0);
            }

            answer.add(temp);
        }

        for (int i = 0; i < row; i++) {
            for (int j = 0; j < column; j++) {

                int previousIndex = i * column + j;

                int newIndex = (previousIndex + k) % total;

                int newRow = newIndex / column;
                int newColumn = newIndex % column;

                answer.get(newRow).set(newColumn, grid[i][j]);
            }
        }

        return answer;
    }
}

File: 1260. Shift 2D Grid.cpp
class Solution {
public:
    vector<vector<int>> shiftGrid(vector<vector<int>>& grid, int k) {
        int row = grid.size();
        int column = grid[0].size();
        int total = row * column;

        k %= total;
        vector< vector < int >> answer(row, vector<int > (column));

        for (int i = 0; i < row; i++) {
            for (int j = 0; j < column; j++) {

                int previous_index = i * column + j;

                int new_index = (previous_index + k) % total;

                int new_row = new_index / column;
                int new_column = new_index % column;

                answer[new_row][new_column] = grid[i][j];
            }
        }

        return answer;
     }
};
